File: challenges/58/solution.ts
var lengthOfLastWord = function (s) {
  let idx = s.length - 1;
  let len = 0;
  while (idx >= 0) {
    if (s[idx] === " ") {
      idx--;
      continue;
    } else break;
  }
  while (idx >= 0) {
    if (s[idx] !== " ") {
      idx--;
      len++;
    } else break;
  }
  return len;
};

File: challenges/58/solution.py
def lengthOfLastWord(s: str) -> int:

    # length of the initial word (starting from the end of the string)
    start = len(s)

    started = False
    for pos in range(len(s)-1, 0, -1):

        c = s[pos]
        
        # starting position
        if c is not " ":
            if not started:
                start = pos
                started = True
                continue

        if started:
            if c is " ":
                return start-pos

    return start+1